### PR TITLE
Fixes #166

### LINKED ISSUE #166: NMS.KK.ACCESS.014 - Rule for marking subscription completed
Kilkari service shall be completed after the Service Pack runs for its scheduled duration (subscription state = Completed).

### PR DESCRIPTION
**Pull request by frankhuster**
The status of a subscription will change to completed during the CDR
processing when the last message has been successfully played (or not)

### CODE REVIEW

File: imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    @Autowired
+    public SubscriptionCompleter(SubscriptionService subscriptionService,
+                                 CallRetryDataService callRetryDataService) {
+        this.subscriptionService = subscriptionService;
+        this.callRetryDataService = callRetryDataService;
+    }
+
+
+    @MotechListener(subjects = { COMPLETE_SUBSCRIPTION })
+    public void completeSubscription(MotechEvent event) {
```
Comment (larubbio) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: Should this live in the IMI module or the kilkari module?

Comment (frankhuster) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: The IMI module I would think.
The CDR class is 'specific' to what we get back from IMI.

Comment (larubbio) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: Sure but don't you put it into an event and then run business logic around
the closing of subscriptions off of it?

Comment (frankhuster) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: Yep.
But that would make Kilkari dependent of IMI.
I guess we could create a Kilkari specific DTO instead.
Should we have the Kilkari module handle all three CDR procesing events, actually?

Comment (larubbio) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: I think so.  I imagine we would define a dto that the IVR interface has to produce and publish.  Then all event handling that result in changes to subscriptions would live in the kilkari module.  If we change ivr providers the new module just needs to recreate and publish those events.

Comment (frankhuster) [inline imi/src/main/java/org/motechproject/nms/imi/component/SubscriptionCompleter.java]: Yep, sounds like a good idea.
Wanna merge this PR and I'll put the IMI => KK refactor in a new PR?

**Comment by frankhuster** [comment]
Hey @larubbio, waiting on your answer for this guy.

**Comment by larubbio** [comment]
I haven't had a chance to look at this beyond that first event listener yet
(I was doing it in between compiles last night and didn't get that far on
it).

**Pull request merged**